### DISCUSSION TITLE
AI Chat with PDF/Word/Text - (Laravel AI Labs)

### DISCUSSION THREAD

**Discussion by adrianmtanase** [laravel/framework#49918] [Show and tell]
This year I've decided to start giving back to this wonderful community, and will be heavily focusing on Open Source work.

My initiative is Laravel AI Labs - a deep dive on all AI related PHP programming with an emphasis on packages.

Throughout this year I'll be demystifying how to create various Web Services leveraging AI, be it through OpenAI's chat GPT or local LLM models. Aiming to create video tutorials and continue creating packages that can ease everyone's transition into AI.

First 2 packages:
- [Laravel Vector Store](https://github.com/adrianmtanase/laravel-vector-store): An abstraction layer for different kinds of Vector Databases. Currently supporting Pinecone.io and Weaviate.
- [Laravel AI File Assistant](https://github.com/LaravelAILabs/file-assistant): Chat with files (PDF, Word, Text) through AI. You can see a [demo repository](https://github.com/LaravelAILabs/file-assistant-demo) here and you can see it in action over at [https://file-assistant.laravelailabs.com/](https://file-assistant.laravelailabs.com/).

What else should I be working on? Please drop a comment here, in order to set some goals and get the ball rolling better.

![cropped](https://github.com/laravel/framework/assets/153947873/a951590b-c23c-43e9-a310-920343e864ee)

**Comment by shaedrich** [comment]
Would it be useful to integrate Vector Databases via Laravel's database functionality or do you think, they are too different from other database engines?

Maybe, Laravel would even benefit from a higher-level `Query` that can be both implemented as `DatabaseQuery` and `HttpQuery`, even `GraphQlQuery` 🤔

**Reply by tntsoft** [reply]
It's an interesting and difficult question. For example Postgres can already handle vectors, and Laravel already has Eloquent for it, and [pgvector](https://github.com/pgvector/pgvector-php) to cast to an instance of a Vector.

Planetscale is working on developing this functionality for MySQL as well, and when it's ready I'll be covering the functionality throught the 'laravel-vector-database' package.

My immediate goal is to lower the complexity bar of programmers trying to breach the AI market by creating powerful packages that can combine into AI applications without much work. For example the 'file-assistant-demo' practically took 10 lines of code for the AI part.

While a sort of Eloquent vector query sounds exciting, it would take me away from my current path for the forseable future, and already having tools like Pinecone.io at our disposal, there is no immediate need for newcomers to host their own in order to get in on the action.

If you do require to host your own, you can use [weaviate](https://github.com/weaviate/weaviate) which my package supports, but has a higher learning curve.
